### ISSUE TITLE
Problem by Identification

### ISSUE THREAD

**Issue by achraf-boussaada** [zafartahirov/fingerpi#18]
Hi everyone !! i know this is only a demo version and not a complete project !  i started working on an authentication system and now i encountered a problem when i try to match a template by the ones already saved in the database of the scanner. To do so i used this snipet of code : 

`		makeTemplateResponse = f.MakeTemplate()`
				`if makeTemplateResponse[0]['ACK'] :`
					`indetifyResponse = f.IdentifyTemplate(makeTemplateResponse)`
					`print indetifyResponse`
so when i execute this it gives me this error : 

> Traceback (most recent call last):
  File "finger.py", line 169, in <module>
    indetifyResponse = f.IdentifyTemplate(makeTemplateResponse)
  File "/home/pi/Desktop/fingerpi/fingerpi/fingerpi.py", line 213, in IdentifyTemplate
    if self.sendData(template, 498):
  File "/home/pi/Desktop/fingerpi/fingerpi/fingerpi.py", line 53, in sendData
    packet = encode_data_packet(data, data_len, device_id = self.device_id)
  File "/home/pi/Desktop/fingerpi/fingerpi/base.py", line 72, in encode_data_packet
    data                 # Data to be sent
struct.error: argument for 's' must be a string

i tried to print only  `makeTemplateResponse[1]['Data'] ` it gives me some weird characters i tried to change the encoding but nothing works. 
As you can see from the following picture when i print the whole response i don't have problem with the data but when i extract it i don't get a valid string.
## Response from creating a template : 
![25-01-2018 22-52-42](https://user-images.githubusercontent.com/31047155/35414471-d9a491aa-0222-11e8-8e60-3dba42bcc764.png)
## Data after extracting it from Response : 
![25-01-2018 22-57-15](https://user-images.githubusercontent.com/31047155/35414582-28687338-0223-11e8-9415-0b677fc9c3c5.png)
## Note : 
Before comming here i tried couple of solutions but nothing works ! any help will be much appreciated

**Comment by zafartahirov** [comment]
That's not how this sensor works:
1. You cannot compare the fingerprints to the ones you have in the database -- this is a different algorithm, and you will have to implement some kind of "fuzzy hash" or Bozorth3 algorithm
2. Getting the data from the sensor returns the template for the fingerprint, not the fingerprint itself for security purposes.
3. If you want to take pictures of the fingerprints, just get a good camera, and good low focal length optics -- it will work much better.

to summarize -- this device is probably not suitable for your case and your case requires much more work than you think

**Comment by ebenezer-isaac** [comment]
> To my understanding to check if there is a match the `data` from the `identifyResponse` is going to be compared with the ones stored in the sensor. What I wanted to achieve is saving only the data field in a `remote database` (so I can save more than 200 fingerprints which is the capacity of the sensor) and every time there is a new fingerprint detected the comparison will happen with the remote database not the one inside the sensor. But I think my approach was incorrect and I ended up using the sensor database. I'm done working on that project but if you have a solution for this topic for future readers please suggest it. Many thanks

@achraf-boussaada
Hi, Im trying to achive the same objective. I have been trying for days together, can you help me out?
Does it still work? How did you make it, I mean what is the workflow and what algorithm did u use? It would be a lot helpful if you could share your code :P
